pkg/cachemanager/cachemanager.go

```diff
@@ -128,8 +129,21 @@ func (c *CacheManager) UpsertSource(ctx context.Context, sourceKey aggregator.Ke
 	// may become unreferenced and need to be deleted; this will be handled async
 	// in the manageCache loop.
 
-	if err := c.replaceWatchSet(ctx); err != nil {
-		return fmt.Errorf("error watching new gvks: %w", err)
+	err := c.replaceWatchSet(ctx)
```

Comment:
I think your desired behavior is (almost) what is happening here:

* replaceWatchSet will always call registrar.ReplaceWatch() with the full set of GVKs-to-be-synced
* registrar.ReplaceWatch() will attempt to watch net-new GVKs, sending a GVK-specific error if any fail.
* interpretErr will filter out any errors that are not associated with a GVK in the current sync set

One bug is that the case where `general` returns true is not properly handed (`failedGVKs` is null in that case, which means the `if general` case never executes.

Reply:
> One bug is that the case where general returns true is not properly handed (failedGVKs is null in that case, which means the if general case never executes).

@maxsmythe iirc, we discussed that if the err is "general", then we would go thru all of the GVKs being watched (not just the ones in the request) and TryCancel expectations for those since that errors would apply to all watched gvks. 

That is why `failedGVKs` comes out as nil since its value is ignored for the general case.

Reply:
Understood. Step through the code path. What happens if `general` is `true` and `faildGVKs` is `nil`?

Right now, UpsertSource() would return `nil`, when it should return the wrapped error.

Reply:
Oh I see the issue now, thank you for being patient 🙏🏼 💯